File: server/src/routes/health.ts
import { Router } from 'express';
import { opGet } from '../openproject/client.js';
import { asyncHandler } from '../middleware/error.js';

const router = Router();

router.get(
  '/',
  asyncHandler(async (_req, res) => {
    try {
      await opGet('/api/v3/users?pageSize=1');
      res.json({ ok: true, op: 'reachable' });
    } catch {
      res.json({ ok: true, op: 'unreachable' });
    }
  }),
);

export default router;

File: server_py/app/routes/health.py
from fastapi import APIRouter

from app.openproject.client import op_get


router = APIRouter(prefix="/api/health")


@router.get("")
async def health() -> dict[str, bool | str]:
    try:
        await op_get("/api/v3/users?pageSize=1")
        return {"ok": True, "op": "reachable"}
    except Exception:
        return {"ok": True, "op": "unreachable"}
